### DISCUSSION TITLE
Docking mock up

### DISCUSSION THREAD

**Discussion by gmptrr98** [iRobotEducation/create3_docs#195] [Questions] [answered]
Hello, I'm trying to build a dock using IR LED and an Arduino. I'm almost able to make it work: subscribing to the /ir_opcode topic I'm able to see the response from Create 3 and I get the correct results, which means that the circuit is working and each LED is recognised by the robot. 
What I don't understand is why, even if I get ir_opcode of 173 (which means that force field, red and green buoy are visible), the /dock topic publishes on dock_visible=FALSE.  Due to this, I'm not able to run the ros action for docking.  

https://user-images.githubusercontent.com/110682193/190193935-02c7ae7e-6edf-4dff-8be6-7efbf5ddf773.mov

Do I need some extra sensors to make my dock recognised? Or is there any way to overwrite the dock_visible to TRUE so that I can run my action?

**Comment by justinIRBT** [comment, marked as answer]
Out of curiosity, if you just produce one of the opcodes, like RED, does the robot report dock_visible?  I don't think our dock produces them all simultaneously.  Do you still have the physical dock that came with the robot to compare ir_opcode behavior on the robot or did you take it apart for your project?

**Reply by gmptrr98** [reply]
I have the real dock, I just noticed that with it, even when very close, it flickers between 161 (Force field) and 172 (Green and Red), I never get 173. I'll make some tests without the force field led and see if that's the problem.
![true_dock_opcode](https://user-images.githubusercontent.com/110682193/190201387-8ef157e9-7017-4d43-b6d0-a401dd3f1a59.jpg)

**Reply by gmptrr98** [reply]
Ok I figured out. If I make field, green and red flash together it doesn't recognize it as a dock (so opcode 165, 169 and 173). For any other opcode it works. 
Thank you!
